### PR TITLE
Add UUID attribute type

### PR DESCRIPTION
**Pull request by NazarioJL**
* Custom attribute is backed by unicode 'S' type in DynamoDB type this allows users to use native UUID types as dynamodb attributes.

### CODE REVIEW

**Comment by NazarioJL** [comment]
PTAL @ikonst

**Comment by NazarioJL** [comment]
🔨

**Comment by ikonst** [comment]
Please merge master. @nolanlum has fixed the test issue.

File: pynamodb_attributes/uuid.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    attr_type = pynamodb.constants.STRING
+
+    def __init__(self, remove_dashes: bool = False, **kwargs: Any) -> None:
+        """
+        Initializes a UUIDAttribute object.
+
+        :param remove_dashes: will remove the dashes in the string
+                              representation if True. Defaults to False.
+        :param kwargs: Extra kwargs passed to base Attribute object.
```
Comment (ikonst) [inline pynamodb_attributes/uuid.py]: I don't think this is needed?

File: pynamodb_attributes/uuid.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    PynamoDB attribute to for UUIDs. These are backed by DynamoDB unicode (`S`) types.
+    """
+
+    attr_type = pynamodb.constants.STRING
+
+    def __init__(self, remove_dashes: bool = False, **kwargs: Any) -> None:
+        """
+        Initializes a UUIDAttribute object.
+
+        :param remove_dashes: will remove the dashes in the string
```
Comment (ikonst) [inline pynamodb_attributes/uuid.py]:
```suggestion
        :param remove_dashes: if set, will string serialization will be without dashes
```

File: pynamodb_attributes/uuid.py (hunk 1 above)
Comment (NazarioJL) [reply pynamodb_attributes/uuid.py]: Needed in the description, or actually as a parameter? How to you express things like, `null=True`?

Comment (ikonst) [reply pynamodb_attributes/uuid.py]: Not needed in the docstring, I meant :)

File: pynamodb_attributes/uuid.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    """
+
+    attr_type = pynamodb.constants.STRING
+
+    def __init__(self, remove_dashes: bool = False, **kwargs: Any) -> None:
+        """
+        Initializes a UUIDAttribute object.
+
+        :param remove_dashes: if set, the string serialization will be without dashes.
+                              Defaults to False.
```
Comment (ikonst) [inline pynamodb_attributes/uuid.py:21]: The default is already part of the signature
```suggestion
```

File: pynamodb_attributes/uuid.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from typing import TYPE_CHECKING
+from uuid import UUID
+
+import pynamodb.constants
+from pynamodb.attributes import Attribute
+
+
+class UUIDAttribute(Attribute):
+    """
+    PynamoDB attribute to for UUIDs. These are backed by DynamoDB unicode (`S`) types.
```
Comment (ikonst) [inline pynamodb_attributes/uuid.py:11]:
```suggestion
    PynamoDB attribute to for UUIDs. These are backed by DynamoDB String (`S`) types.
```

**Comment by NazarioJL** [comment]
💨

**Comment by ikonst** [comment]
@NazarioJL (1) fix tests, (2) this is an OSS repo so it doesn't use Lyft's Submit-Queue :)

**Comment by ikonst** [comment]
oops!

**Review by ikonst — APPROVED** [review]
Just fix the markdown spacing with `pre-commit`

**Pull request merged**